criu/cgroup.c

```diff
@@ -1208,32 +1207,27 @@ static int move_in_cgroup(CgSetEntry *se, bool setup_cgns)
 	return 0;
 }
 
-int prepare_task_cgroup(struct pstree_item *me)
+int prepare_task_cgroup(u32 cg_set, u32 parent_cg_set)
```

Comment:
Instead of pushing code out of this into its parent, why not extract out another function, say `prepare_cgroup` which is called by `prepare_task_cgroup`.

With the changes as they are now, the calling code in `restore_task_in_children` is getting polluted.

Anyway, this is not important now, come back to it when you have end-to-end working implementation.